tools/bin/mbedtls-archive-epic.py

```diff
@@ -0,0 +1,340 @@
+import sys
+import json
+import requests
+from shlex import split
+from string import Template
+from subprocess import Popen, check_output, PIPE
+from pprint import pprint
+
+token = os.environ['GITHUB_TOKEN']
+
+
+######################### User Config ###########################
+# Instuctions
+# - Please ensure that your github cli token is present in your enviroment
+# - Set the epic name to tranfer here.
+# - Create the epic column to the arhive board, matching the src name.
+# - If needed adjust the closed_after date to be in a resonable window
+# - Fine tune max entries selection if some entries are missing.
+# - Private issues and drafts are not picked up by the script.
+
+epic = "3.6.1 patch release"       # Name of epic that needs to be moved
+                                    # (It needs to match an empty epic on v2 board)
+closed_after_date = "2024-01-01"    # Only include entries closed after that date
+
+max_entries = 500                   # How many maximum entries the backend will populate.
+debug = True                       # Set to True to print out the github cli commands.
+
+############################# Constants #############################
+"""
+The following values should remain the same across the project.
+If need to modify them, they can querried as follows:
+
+gh project list --owner Mbed-TLS number 13
+    NUMBER  TITLE                      STATE  ID
+    13      Past EPICs                 open   PVT_kwDOBcuPHc4AgPo7
+
+gh project --owner Mbed-TLS field-list 13
+    NAME                  DATA TYPE                   ID
+    Status                ProjectV2SingleSelectField  PVTSSF_lADOBcuPHc4AgPo7zgVai1o
+
+"""
+project_epics_for_mbedtls = "PVT_kwDOBcuPHc4AnF0W"
+project_past_epics = "PVT_kwDOBcuPHc4AgPo7"
+project_field_status = "PVTSSF_lADOBcuPHc4AgPo7zgVai1o"
+project_field_owner = "Mbed-TLS"
+
+projects_repo = "Mbed-TLS/mbedtls"
+active_project = "EPICs for Mbed TLS"
+archive_project = "Past EPICs"
+archive_project_board_no = "13"
+
+############################ Github Cli gap #########################
+""" This snippet is covering functionality missing for the current version of github cli.
+    We need to querry the node-id for the Status SingleSelectField, since that is used as
+    epic name columns. """
+
+gql_get_ids_querry = Template(
+    """
+  query{
+  node(id: "$project_uid") {
+    ... on ProjectV2 {
+      fields(first: 100) {
+        nodes {
+          ... on ProjectV2SingleSelectField {
+            name options {
+              id
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+}"""
+)
+
+########################### Helpers #################################
+
+class ShellExecResult:
+
+    """
+    Class to encapsulate the returns from a shell execution.
+    """
+
+    def __init__(self, success, stdout, stderr):
+
+        self.success = success
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+def do_shell_exec(exec_string, expected_result=0):
```

Comment:
Shouldn't this just be `subprocess.check_output`?

* I don't see this being used for the error cases, except to print additional error messages that aren't particularly helpful.
* Sometimes the caller forgets to check the return code, it would be better to keep using an exception on failure.
* Callers don't actually rely on having a shell, they're just passing arguments to one program, sometimes with dodgy quoting. So it would be easier to not use a shell.